src/openrct2/config.c

```diff
@@ -544,6 +544,17 @@ bool config_open(const utf8string path)
 
 	free(lineBuffer);
 	SDL_RWclose(file);
+
+	// Window limit must be kept within valid range.
+	if (gConfigGeneral.window_limit < WINDOW_LIMIT_MIN) {
+		gConfigGeneral.window_limit = WINDOW_LIMIT_MIN;
+		config_save_default();
```

Comment:
why would we not want it to update the config file? wouldn't being notified that the game didn't accept a specific value be useful? like, if i set a value to 3, and the game doesn't accept 3, i wouldn't know for sure if my file was still saying 3 yet it was showing 4 windows. however if it updated it to 4 ande i seen it next time i go to modify the file, i couldn think "oh so whatever i just typed wasnt accepted"